Add vertical scrollable orientation

moonboots opened the pull request:
to: @majapw 

Adds a new orientation that makes `DayPicker` vertically scrollable instead of using pagination. This scrollable `DayPicker` will fill the height of its parent and show `numberOfMonths`.

Replaces this [previous PR](https://github.com/airbnb/react-dates/pull/244). I've separated css by component and threaded the new orientation property throughout.

moonboots commented:
@majapw I've update this PR with a new PropType shape `ScrollableOrientationShape` that's used by internal classes. Top level components that don't support scroll like `DateRangePicker` continue to use `OrientationShape`, which will guard against passing in unsupported orientations.

majapw commented on css/CalendarMonth.scss:
```diff
@@ -35,6 +35,10 @@
   // necessary to not hide borders in FF
   margin-bottom: 2px;
+
+  .CalendarMonth--vertical-scrollable & {
```
I feel like this is not terribly clear? I'd rather have 
```
.CalendarMonth--horizontal.CalendarMonth__caption,
.CalendarMonth--vertical.CalendarMonth__caption {
  padding: 15px 0 35px;
}

.CalendarMonth--vertical-scrollable.CalendarMonth__caption {
  padding: 5px 0;
}
```

majapw commented on src/components/DayPicker.jsx:
```diff
@@ -301,15 +316,20 @@ export default class DayPicker extends React.Component {
 
+    const onNextMonthClick = orientation === VERTICAL_SCROLLABLE
+      ? this.multiplyScrollableMonths
+      : this.onNextMonthClic;
```
`this.onNextMonthClick`, you're missing a k.

Also we prefer an `if... then` over a multiline ternary.

majapw commented on src/components/DayPicker.jsx:
```diff
@@ -376,9 +401,12 @@ export default class DayPicker extends React.Component {
     const dayPickerClassNames = cx('DayPicker', {
       'DayPicker--horizontal': this.isHorizontal(),
       'DayPicker--vertical': this.isVertical(),
+      'DayPicker--vertical-scrollable': this.props.orientation === VERTICAL_SCROLLABLE,
```
I assume you mean to be reused the `verticalScrollable` from line 404

majapw commented on stories/DateRangePicker.js:
```diff
@@ -2,7 +2,11 @@ import React from 'react';
+  ANCHOR_RIGHT,
+  VERTICAL_ORIENTATION,
+  VERTICAL_SCROLLABLE,
+} from '../constants';
```
Did you mean to make this addition?

majapw commented on src/components/DayPickerRangeController.jsx:
```diff
@@ -36,10 +36,11 @@ const propTypes = {
   enableOutsideDays: PropTypes.bool,
   numberOfMonths: PropTypes.number,
-  orientation: OrientationShape,
+  orientation: ScrollableOrientationShape,
```
Is this correct?

moonboots commented:
Thanks @majapw, fixed up the PR. The orientation changes in `DayPickerRangeController` allow a custom `DayRangePicker` to pass the scrollable orientation to the underlying `DayPicker`. There were a few leftover `scrollable` props from my previous PR that I've removed.

oyeanuj commented:
@majapw @moonboots Curious if there was a reason that this orientation wasn't included for `SingleDatePicker`? 

In dealing with responsive decisions, this might be a much useful option.
